src/core/main.c

```diff
@@ -2064,6 +2061,9 @@ int main(int argc, char *argv[]) {
                 arg_serialization = safe_fclose(arg_serialization);
                 fds = fdset_free(fds);
 
+                /* Reopen the console */
+                (void) make_console_stdio();
```

Comment:
Are you sure we want to restore make_console_stdio() in the case we're reexecuting /bin/init ?

I mean whatever is behind /bin/init, we're likely to see the same issue.